lib/core/src/irods_server_properties.cpp

```diff
@@ -22,11 +32,127 @@ namespace irods
+				}
+
+				string = static_cast<string_t *>(stream);
+
+				new_len = string->len + size*nmeb;
+
+				/* Reallocate memory with space for new content */
+				/* Add an extra byte for terminating null char */
+				tmp_ptr = realloc(string->ptr, new_len + 1);
+				if (!tmp_ptr) {
+						rodsLog(LOG_ERROR, "%s", "irodsCurl::write_string: realloc failed.");
+						return -1;
+				}
+
+				string->ptr = (char*)tmp_ptr;
+
+				/* Append new content to string and terminating '\0' */
+				memcpy(string->ptr + string->len, ptr, size*nmeb);
+				string->ptr[new_len] = '\0';
+				string->len = new_len;
+
+				return size*nmeb;
+		} // curl_write_str
+
+        auto use_remote_configuration() -> bool
+        {
+            return getenv(ENDPOINT_KW.c_str()) != nullptr;
+        }
+
+        auto capture_remote_configuration() -> json
+        {
+            auto* url = getenv(ENDPOINT_KW.c_str());
+
+            if(!url) {
+                THROW(SYS_INTERNAL_ERR,
+                      "configuation endpoint environment variable is not set");
+            }
+
+            CURLcode res = CURLE_OK;
+            string_t string;
+
+            // Destination string_t init
+            string.ptr = strdup("");
+            string.len = 0;
+
+            CURL* curl = curl_easy_init();
+
+            // Set up easy handler
+            curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcurl-agent/1.0");
+            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, curl_write_str);
+            curl_easy_setopt(curl, CURLOPT_WRITEDATA, &string);
+            curl_easy_setopt(curl, CURLOPT_URL, url);
+
+            // CURL call
+            res = curl_easy_perform(curl);
+
+            curl_easy_cleanup(curl);
+
+            // Error logging
+            if ( res != CURLE_OK ) {
+                THROW(SYS_INTERNAL_ERR, curl_easy_strerror(res));
+            }
+
+            auto cfg_json_str = std::string{string.ptr, string.ptr+string.len};
+
+            free(string.ptr);
+
+            return json::parse(cfg_json_str);
+
+        } // capture_remote_configuration
+
+        auto process_remote_configuration(const json& cfg) -> json
+        {
+            if(!cfg.contains("server_config.json")) {
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      "remote configuration missing server_config.json contents");
+            }
+
+            auto tmp = cfg.at("server_config.json");
+
+            if(cfg.contains("database_config.json")) {
+                for(auto [k, v] : cfg.items()) {
+                    tmp[k] = v;
+                }
+            }
+
+            if(cfg.contains("hosts_config.json")) {
+                tmp[irods::HOSTS_CONFIG_JSON_OBJECT_KW] = cfg.at("hosts_config.json");
+            }
+
+            return tmp;
+
+        } // process_remote_configuration
+
+    } // namespace
+
 
     server_properties& server_properties::instance()
     {
-        static server_properties singleton;
+        static server_properties singleton{};
```

Comment:
If the type has a default constructor, then the curly braces aren't needed. Just an fyi.